pkg/klusterlet/clusterinfo/clusterinfo_controller.go

```diff
@@ -386,6 +390,32 @@ func (r *ClusterInfoReconciler) getCloudVendor() clusterv1beta1.CloudVendorType
 	return clusterv1beta1.CloudVendorOther
 }
 
+// this API is created for the Openshift Dedicated platform to manage various permissions.
+// defined in https://github.com/openshift/rbac-permissions-operator/blob/master/pkg/apis/managed/v1alpha1/subjectpermission_types.go
+var subjectpermissionsGVR = schema.GroupVersionResource{
+	Group:    "managed.openshift.io",
+	Version:  "v1alpha1",
+	Resource: "subjectpermissions",
+}
+
+const (
+	dedicatedAdmin    = "dedicated-admins"
+	rbacPermissionsNs = "openshift-rbac-permissions"
+)
+
+func (r *ClusterInfoReconciler) isOpenShiftDedicated() bool {
+	_, err := r.ManagedClusterDynamicClient.Resource(subjectpermissionsGVR).
```

Comment:
could we just check the existence of this api resources? With this, you would need additional rbac

Reply:
yes. modified to use group name `managed.openshift.io` to detect if the cluster is osd, like using group name `project.openshift.io `to detect if the cluster is OCP.